logstash-core/src/main/java/org/logstash/common/io/ByteArrayStreamOutput.java

```diff
@@ -0,0 +1,65 @@
+package org.logstash.common.io;
+
+import java.nio.ByteBuffer;
+
+public class ByteArrayStreamOutput extends StreamOutput {
+    private byte[] bytes;
+
+    private int pos;
```

Comment:
I think it would be clearer for this variable to be named `position` since its getter is `getPosition`